File: app/src/main/java/org/supla/android/ChannelCaptionEditor.java
package org.supla.android;

import android.content.Context;

import org.supla.android.db.Channel;
import org.supla.android.db.DbHelper;
import org.supla.android.db.Location;
import org.supla.android.lib.SuplaClient;

public class ChannelCaptionEditor extends CaptionEditor {
    public ChannelCaptionEditor(Context context) {
        super(context);
    }

    @Override
    protected int getTitle() {
        return R.string.channel_name;
    }

    @Override
    protected String getCaption() {
        DbHelper dbH = DbHelper.getInstance(getContext());
        Channel channel = dbH.getChannel(getId());
        if (channel != null && channel.getCaption() != null) {
            return channel.getCaption();
        }
        return "";
    }

    @Override
    protected void applyChanged(String newCaption) {
        DbHelper dbH = DbHelper.getInstance(getContext());
        Channel channel = dbH.getChannel(getId());
        if (channel != null) {
            channel.setCaption(newCaption);
            dbH.updateChannel(channel);

            SuplaClient client = SuplaApp.getApp().getSuplaClient();
            if (client != null) {
                client.setChannelCaption(getId(), newCaption);
            }
        }
    }

    @Override
    protected int getHint() {
        return R.string.str_default;
    }
}

File: app/src/main/java/org/supla/android/ui/dialogs/ChannelCaptionEditor.kt
package org.supla.android.ui.dialogs

import android.content.Context
import org.supla.android.CaptionEditor
import org.supla.android.R
import org.supla.android.SuplaApp
import org.supla.android.db.DbHelper

class ChannelCaptionEditor(context: Context?) : CaptionEditor(context) {

  var captionChangedListener: () -> Unit = { }

  override fun getTitle(): Int {
    return R.string.channel_name
  }

  override fun getCaption(): String {
    val dbH = DbHelper.getInstance(context)
    val channel = dbH.getChannel(id)
    return if (channel != null && channel.caption != null) {
      channel.caption
    } else ""
  }

  override fun applyChanged(newCaption: String) {
    val dbH = DbHelper.getInstance(context)
    val channel = dbH.getChannel(id)
    if (channel != null) {
      channel.caption = newCaption
      dbH.updateChannel(channel)
      val client = SuplaApp.getApp().getSuplaClient()
      client?.setChannelCaption(id, newCaption)
    }
    captionChangedListener()
  }

  override fun getHint(): Int {
    return R.string.str_default
  }
}